lib/wallaroo/core/sink/tcp_sink/tcp_sink.pony

```diff
@@ -383,9 +391,10 @@ actor TCPSink is Sink
       recover val Array[ByteSeq] end)
 
   be prepare_for_rollback() =>
-    finish_preparing_for_rollback()
+    let dummy_token = CheckpointBarrierToken(0)
+    finish_preparing_for_rollback(dummy_token)
 
-  fun ref finish_preparing_for_rollback() =>
+  fun ref finish_preparing_for_rollback(token: BarrierToken) =>
```

Comment:
Let's change this interface to take a (BarrierToken | None) so we can eliminate the dummy token. This is infrequent so shouldn't have a meaningful perf impact.

Reply:
Updated by commit 8db0273